src/dcqc/file.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import os
+import re
+from collections.abc import Collection, Mapping
+from copy import deepcopy
+from dataclasses import asdict, dataclass
+from pathlib import Path
+from typing import Any, Optional
+
+from dcqc.mixins import SerializableMixin
+from dcqc.utils import open_parent_fs
+
+
+@dataclass
+class FileType:
+    # Class attributes
+    # A type hint is omitted so this attribute isn't
+    # picked up by @dataclass as an instance attribute
+    _registry = dict()  # type: ignore
+
+    # Instance attributes
+    name: str
+    file_extensions: tuple[str, ...]
+
+    def __init__(self, name: str, file_extensions: Collection[str]):
+        self.name = name
+        self.file_extensions = tuple(file_extensions)
```

Comment:
Here, I had to override the default `__init__()` method because I needed to run `register_file_type()` as part of initialization. Also, since the `str.endswith()` method only works with tuples, I wanted to cast any sequence of strings to a tuple. The different in type hints is that internally, the `file_extensions` attribute will be stored as a tuple, but you can provide any sequence of file extensions to the class.